src/dune_rules/compilation_context.ml

```diff
@@ -270,13 +287,18 @@ let for_module_generated_at_link_time cctx ~requires ~module_ =
        their implementation must also be compiled with -opaque *)
     Ocaml.Version.supports_opaque_for_mli cctx.ocaml.version
   in
+  let requires_compile =
```

Comment:
```suggestion
  let requires =
```

Reply:
Let's avoid unnecessary diffs.

Reply:
The problem here is that now `requires_link` and `requires_compile` are of different types, and in the old dune the list `requires` was passed to both fields. Thats why I named the list with bools as `requires_compile`. I will think of another way to reduce the diff.